src/processor.cpp

```diff
@@ -0,0 +1,401 @@
+#include "processor.h"
+#include "game.h" // TODO: This is a circular dependency //it was for gates, probably will be for this too
```

Comment:
To avoid circular dependencies, do not include headers in headers. That is, `game.h` should not touch `processor.h` at all, neither should `processor.h` touch `game.h`.
Including both headers in both `.cpp` files is fine and won't result in conflicts.